src/main/java/ua.horobets/hw23/servlet/CarServlet.java

```diff
@@ -0,0 +1,95 @@
+package ua.horobets.hw23.servlet;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import ua.horobets.hw23.dao.CarDao;
+import ua.horobets.hw23.model.Car;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@WebServlet("/car/*")
+public class CarServlet extends HttpServlet {
+    private final CarDao carDao = CarDao.getInstance();
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String contextPath = req.getPathInfo();
+        resp.setContentType("application/json");
+
+        if (contextPath == null || contextPath.equals("/")) {
+            List<Car> allCars = carDao.getAllCar();
+            String json = objectMapper.writeValueAsString(allCars);
+            resp.getWriter().println(json);
+        } else {
+            String[] pathParts = contextPath.split("/");
+            if (pathParts.length == 2) {
+                try {
+                    int carId = Integer.parseInt(pathParts[1]);
```

Comment:
This all logic that retrieve ID from context path we can move to another service (or at least a method, but, if we are going to have other servlets that needs to retrieve id? So, create some util class that will be responsible for that).

And use it in delete method, because in doDelete you need ID as well